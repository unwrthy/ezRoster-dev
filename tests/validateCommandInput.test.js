const validateCommandInput = require('../src/validateCommandInput')

test('It should return void if the validation pass', () => {
  expect(validateCommandInput(['url1', 'url2'], 'somedomain', ['slashtag1'])).toBeUndefined()
})

test('It should return an error if no url is specified', () => {
  expect(validateCommandInput()).toMatchSnapshot()
})

test('It should return an error if multiple domains are provided', () => {
  expect(validateCommandInput(['url1', 'url2'], ['domain1'])).toMatchSnapshot()
})

test('It should return an error if there are more slashtags than urls', () => {
  expect(validateCommandInput(['url1', 'url2'], 'somedomain', ['slashtag1', 'slashtag2', 'slashtag3'])).toMatchSnapshot()
})

test('It should return an error if there are more than 5 urls', () => {
  expect(validateCommandInput([
    'http://fstack.link/10',
    'http://fstack.link/11',
    'http://fstack.link/12',
    'http://fstack.link/13',
    'http://fstack.link/14',
    'http://fstack.link/15'
  ])).toMatchSnapshot()
})
