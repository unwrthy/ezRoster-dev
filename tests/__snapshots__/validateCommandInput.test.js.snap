// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should return an error if multiple domains are provided 1`] = `[Error: Multiple domains found. You can specify at most one domain]`;

exports[`It should return an error if no url is specified 1`] = `[Error: No url found in the message]`;

exports[`It should return an error if there are more slashtags than urls 1`] = `[Error: Urls/Slashtags mismatch: you specified more slashtags than urls]`;

exports[`It should return an error if there are more than 5 urls 1`] = `[Error: You cannot shorten more than 5 URLs at the time]`;
